tools/checky/src/checks.rs

```diff
@@ -0,0 +1,15 @@
+use std::ops::Range;
+
+use crate::rust_file::RustFile;
+
+pub mod empty_lines;
+pub mod mod_use_order;
+
+pub type Report<'a> = ariadne::Report<'a, (&'a str, Range<usize>)>;
+
+pub fn check(file: &RustFile) -> Vec<Report> {
+    let mut reports = Vec::new();
+    reports.extend(empty_lines::check(file));
```

Comment:
Suggestion for the check API, as some check may want to have configuration parameters:
```
pub trait Check {
  fn check(&self) -> impl IntoIterator<Item = Report>;
}

// [...]

let empty_lines = EmptyLines::new(/*possible setup here, read from toml etc./*);
reports.extend(empty_lines.check(file));
```

or maybe even additionally

```
pub trait Check {
  fn check(&self, context: &mut ReportContext) -> impl IntoIterator<Item = Report>;
}
```
which hides the allocation in `Vec<Report>` or even directly prints them. This way the check only calls something like `context.report(...)`

What do you think?

Reply:
I tried quickly but this requires to think a bit about lifetimes etc. which is too much for me in this PR. Either someone takes this over or we could do this in a next iteration.